### ISSUE TITLE
More accurate recruit list accessed via hot-key

### ISSUE THREAD

**Issue by doofus-01** [wesnoth/wesnoth#1911]
(This issue/wish is for BfW 1.13.8+, but is not specific to that version)
If a side has two leaders, and they have access to different recruit list (via the extra_recruit tags), the combined recruit list is shown if the menu is called up with hot-key.  If one of the leaders is on a keep <b>and</b> is moused-over, only the available units are shown in the recruit list menu.

I'm not sure what the best behaviour would be, but doubt it is the current behaviour.
Two possibilities that wouldn't change the current recruit menu:
1.  Show only (side-wide) recruits by default, only show extra_recruits if actually available (leader on keep & moused-over).
2.  Show only recruits available from the last leader moused-over.

Better might be to show all recruits, but grey out the unavailable units, or put them in a different list.

This might be related (not sure if it is relevant though, if fendrin is not active)
https://github.com/wesnoth/wesnoth/issues/1283

**Comment by CelticMinstrel** [comment]
I think there are three basic cases:

1. Mouse is not over a keep or castle tile. In that case, all recruits that can be recruited should be shown as available. If you have two leaders but only one is on a keep, you should only be able to select that leader's recruits. If both are on a keep, but one has no connected free castle tiles, you should only be able to select the other leader's recruits.

2. Mouse is over a castle tile, or an unoccupied keep tile. In that case, you should be able to select only units recruitable by leaders that are on a keep connected to that castle (or keep) tile.

3. Mouse is over a leader on a keep. In that case, you should be able to select only units recruitable by that leader.

(Note: When I say "shown as available" I mean listed and enabled; I won't rule out others being listed but disabled.)

**Comment by gfgtdf** [comment]
I actually do think the current bahviour is good, afaik currently there are 2 cases:
1) you have the mouse over a tile that you can recruit on, then place the recruit on that hex and show only units that can be recruited on that hex.
2) you have the mouse over a tile that can not be recruiuited on, then show all recruits that can be recruited on any hex and place the recuits on any avaialbe hex.

**Comment by doofus-01** [comment]
I think CelticMinstrel and I are on the same page, I disagree with gfgtdf.

>2.  you have the mouse over a tile that can not be recruiuited on, then show all recruits that can be recruited on any hex and place the recuits on any avaialbe hex.

That's not a good description of what can happen, with the current behaviour.  
Consider:

- Two leaders on one side, each leader has an extra_recruit list.
- One leader is on a keep, one is not.  
- With mouse over some random non-castle hex, you call up the recruit menu -> you are shown all recruits from all lists.  
- You select some units, they get recruited, but select others and you get a popup message.  There is no indication of which will lead to what.  

>Also this is unrelated to whether the recuit dialog is accessed vai hotkey or context menu.
 
No, hot-key can access the recruit menu when context menu option is unavailable.

**Comment by CelticMinstrel** [comment]
> No, hot-key can access the recruit menu when context menu option is unavailable.

Really? That seems like a separate bug, then.

**Comment by doofus-01** [comment]
Yes, but please don't remove that feature.  Requiring more mouse positioning would be annoying.

**Comment by CelticMinstrel** [comment]
I was thinking of the opposite - if it's accessible by hotkey it should also be in the context menu.

**Comment by gfgtdf** [comment]
hmm i just tested this, and i notices something unrelated: when i open the recruitlist via hotkey i get soem sound notification but if i do this via hotkey i don't. Is this expected bahviour?

**Comment by doofus-01** [comment]
The recruit option disappearing from the context menu seems like it must be deliberate, but that doesn't mean it shouldn't be changed.  I don't know ...
I never noticed the sound notification, I don't know what is better for that either.

**Comment by gfgtdf** [comment]
hmm i just tested again and i cannot reproduce the sound this time, but i was sure it was there, also it was a typical windows sound that'd probably even appear if wesnoths sound was swicthed off.. not sure.

**Comment by Wedge009** [comment]
I'm probably wrong but this sounds somewhat related to #1282. At the very least the multiple leader scenario is mentioned: https://github.com/wesnoth/wesnoth/issues/1282#issuecomment-528354802
